Add alumni components

akshatshah21 opened the pull request:
### Adds an Alumni Page

1. `AlumniPage` is the main component. It consists of a tabbed layout for different years.
2. `AlumniCard` is a card for one individual alumnus. Fake data for now
3. `MobileMenu` is a general menu utility to replace the tabbed layout on smaller screens

![image](https://user-images.githubusercontent.com/47059039/105178323-dccd8d00-5b4d-11eb-91f7-9e057908d95a.png)
![image](https://user-images.githubusercontent.com/47059039/105178450-0e465880-5b4e-11eb-8789-0ce3cfb1b025.png)
![image](https://user-images.githubusercontent.com/47059039/105178492-1900ed80-5b4e-11eb-993b-87f6f9ce7cca.png)

The image mask is a circle right now, but the following are also easily available (or any mask, using [this](https://bennettfeely.com/clippy/) great tool)

![image](https://user-images.githubusercontent.com/47059039/105178767-7d23b180-5b4e-11eb-86f4-694d10862839.png)
![image](https://user-images.githubusercontent.com/47059039/105178855-93ca0880-5b4e-11eb-87e0-30c8b3736d56.png)

The `AlumniCard` doesn't have links or any interactivity yet. I've kept them as plain white `Card`s for now.

Please let me know how this looks.

ShubhankarKG commented:
The layout looks good to me, I guess a 3 column grid would look even better. Tagging @Saif807380 as well for this. Since these are cards, they should have external links and some interactivity. But we'll get to that once we reach a conclusion.

Saif807380 commented:
I agree with @ShubhankarKG. A 3 column grid with slightly larger cards would look even better. No changes apart from this. Great Work!

ShubhankarKG commented on new_client/src/App.js:
```diff
@@ -70,6 +71,7 @@ function App() {
               <ProtectedRoute path="/addevent" component={AddEvent} />
               <ProtectedRoute path="/event/edit/:id" component={AddEvent} />
               <Route path="/resources" component={ResourcePage} />
+              <Route path="/alumni" component={AlumniPage} />
```
I know this isn't complete, but we use `lazy` for code splitting now, so maybe in a later commit, we can update this as well.

akshatshah21 commented:
Updates:
1. The cards flip on click:
![Alumni Page Desktop](https://user-images.githubusercontent.com/47059039/106146189-64457c80-619c-11eb-81f5-64b8cbb8028e.gif)
![Alumni Mobile](https://user-images.githubusercontent.com/47059039/106146195-660f4000-619c-11eb-9782-5d6f10093551.gif)
2. AlumniPage is now a lazy component.
3. We can add icons for different kinds of links, but that will require changing the `Alumnus` model so that it can have multiple links, like Github, LinkedIn, et cetera.

akshatshah21 commented:
Updates:
1. Added social icons 
2. Changed the Alumnus Schema
![Social Icons](https://user-images.githubusercontent.com/47059039/106260197-22711080-6246-11eb-9c64-223e07f59f53.gif)

ShubhankarKG commented:
It seems we are missing the Alumni header link on the Navbar. Please add that as well.

ShubhankarKG reviewed (changes requested):
Rest of the stuff looks good! Just a small suggestion. Thanks for this.

ShubhankarKG reviewed (approved):
The changes look good to me. Thanks!
